src/rpcblockchain.cpp

```diff
@@ -4218,21 +4153,30 @@ Value listitem(const Array& params, bool fHelp)
     }
     else if (sitem == "projects") 
     {
-        for(map<string,StructCPID>::iterator ii=mvBoincProjects.begin(); ii!=mvBoincProjects.end(); ++ii) 
+        for (const auto item : AppCacheFilter("project"))
```

Comment:
You can use const& here to avoid copying. Though I think the appcache filter just returns an iterator pair so it's probably a cheap operation anyway.